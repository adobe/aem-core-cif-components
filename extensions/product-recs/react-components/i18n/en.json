{
  "productrecs:add-to-cart": [
    {
      "type": 0,
      "value": "Add to Cart"
    }
  ],
  "productrecs:add-to-wishlist": [
    {
      "type": 0,
      "value": "Add to Wish List"
    }
  ],
  "productrecs:price-from": [
    {
      "type": 0,
      "value": "from"
    }
  ]
}