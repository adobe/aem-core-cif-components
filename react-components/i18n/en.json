{
  "account:account-created": [
    {
      "type": 0,
      "value": "Account created"
    }
  ],
  "account:account-created-title": [
    {
      "type": 0,
      "value": "Your account was successfully created"
    }
  ],
  "account:account-details-sign-in-text": [
    {
      "type": 0,
      "value": "Please sign in to see the account details."
    }
  ],
  "account:account-icon-text-greeting": [
    {
      "type": 0,
      "value": "Hi, "
    },
    {
      "type": 1,
      "value": "name"
    }
  ],
  "account:account-icon-text-sign-in": [
    {
      "type": 0,
      "value": "Sign In"
    }
  ],
  "account:account-information": [
    {
      "type": 0,
      "value": "Account Information"
    }
  ],
  "account:add-an-address": [
    {
      "type": 0,
      "value": "Add an address"
    }
  ],
  "account:address-book": [
    {
      "type": 0,
      "value": "Address Book"
    }
  ],
  "account:address-book-sign-in-text": [
    {
      "type": 0,
      "value": "Please Sign in to see your address book."
    }
  ],
  "account:address-default-tag": [
    {
      "type": 0,
      "value": "Default"
    }
  ],
  "account:address-delete": [
    {
      "type": 0,
      "value": "Delete"
    }
  ],
  "account:address-delete-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:address-delete-confirm": [
    {
      "type": 0,
      "value": "Delete"
    }
  ],
  "account:address-edit": [
    {
      "type": 0,
      "value": "Edit"
    }
  ],
  "account:address-form-heading": [
    {
      "type": 0,
      "value": "Address"
    }
  ],
  "account:change-password": [
    {
      "type": 0,
      "value": "Change Password"
    }
  ],
  "account:change-password-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:confirm-new-password": [
    {
      "type": 0,
      "value": "Confirm New Password"
    }
  ],
  "account:confirm-password": [
    {
      "type": 0,
      "value": "Confirm Password"
    }
  ],
  "account:continue-shopping": [
    {
      "type": 0,
      "value": "Continue Shopping"
    }
  ],
  "account:create-account": [
    {
      "type": 0,
      "value": "Create account"
    }
  ],
  "account:create-account-lead": [
    {
      "type": 0,
      "value": "Check out faster, use multiple addresses, track orders and more by creating an account!"
    }
  ],
  "account:create-an-account": [
    {
      "type": 0,
      "value": "Create an Account"
    }
  ],
  "account:create-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:create-submit": [
    {
      "type": 0,
      "value": "Submit"
    }
  ],
  "account:current-password": [
    {
      "type": 0,
      "value": "Current Password"
    }
  ],
  "account:edit-account-info": [
    {
      "type": 0,
      "value": "Edit account information"
    }
  ],
  "account:email": [
    {
      "type": 0,
      "value": "Email address"
    }
  ],
  "account:email-confirmation-info": [
    {
      "type": 0,
      "value": "You will receive a link at "
    },
    {
      "type": 1,
      "value": "email"
    },
    {
      "type": 0,
      "value": ". Access that link to confirm your email address."
    }
  ],
  "account:firstname": [
    {
      "type": 0,
      "value": "First Name"
    }
  ],
  "account:forget-password-confirmation": [
    {
      "type": 0,
      "value": "If there is an account associated with "
    },
    {
      "type": 1,
      "value": "email"
    },
    {
      "type": 0,
      "value": ", you will receive an email with a link to change your password."
    }
  ],
  "account:forgot-password": [
    {
      "type": 0,
      "value": "Forgot Password?"
    }
  ],
  "account:forgot-password-cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "account:forgot-password-submit": [
    {
      "type": 0,
      "value": "Submit"
    }
  ],
  "account:lastname": [
    {
      "type": 0,
      "value": "Last Name"
    }
  ],
  "account:my-account": [
    {
      "type": 0,
      "value": "My account"
    }
  ],
  "account:name": [
    {
      "type": 0,
      "value": "Name"
    }
  ],
  "account:new-password": [
    {
      "type": 0,
      "value": "New Password"
    }
  ],
  "account:new-password-confirm": [
    {
      "type": 0,
      "value": "Confirm new Password"
    }
  ],
  "account:password": [
    {
      "type": 0,
      "value": "Password"
    }
  ],
  "account:password-recovery": [
    {
      "type": 0,
      "value": "Password recovery"
    }
  ],
  "account:password-was-changed": [
    {
      "type": 0,
      "value": "Your password was changed."
    }
  ],
  "account:reset-password": [
    {
      "type": 0,
      "value": "Reset Password"
    }
  ],
  "account:reset-password-error": [
    {
      "type": 0,
      "value": "Could not reset password."
    }
  ],
  "account:reset-password-instructions": [
    {
      "type": 0,
      "value": "Enter your email below to receive a password reset link."
    }
  ],
  "account:reset-password-lead": [
    {
      "type": 0,
      "value": "Please choose a new password to complete the password reset."
    }
  ],
  "account:reset-password-missing-token": [
    {
      "type": 0,
      "value": "Missing or invalid token."
    }
  ],
  "account:reset-password-success": [
    {
      "type": 0,
      "value": "Your password was changed. Please log in with your new password."
    }
  ],
  "account:sign-in": [
    {
      "type": 0,
      "value": "Sign In"
    }
  ],
  "account:sign-in-form-title": [
    {
      "type": 0,
      "value": "Sign-in to Your Account"
    }
  ],
  "account:sign-out": [
    {
      "type": 0,
      "value": "Sign Out"
    }
  ],
  "account:signing-in": [
    {
      "type": 0,
      "value": "Signing In"
    }
  ],
  "account:subscribe-news": [
    {
      "type": 0,
      "value": "Subscribe to news and updates"
    }
  ],
  "cart:apply-coupon": [
    {
      "type": 0,
      "value": "Apply Coupon"
    }
  ],
  "cart:continue-shopping": [
    {
      "type": 0,
      "value": "Continue Shopping"
    }
  ],
  "cart:edit-item": [
    {
      "type": 0,
      "value": "Edit item"
    }
  ],
  "cart:enter-coupon": [
    {
      "type": 0,
      "value": "Enter your code"
    }
  ],
  "cart:fetching-data": [
    {
      "type": 0,
      "value": "Fetching cart data..."
    }
  ],
  "cart:itemsWithCount": [
    {
      "offset": 0,
      "options": {
        "one": {
          "value": [
            {
              "type": 7
            },
            {
              "type": 0,
              "value": " item"
            }
          ]
        },
        "other": {
          "value": [
            {
              "type": 7
            },
            {
              "type": 0,
              "value": " items"
            }
          ]
        }
      },
      "pluralType": "cardinal",
      "type": 6,
      "value": "count"
    }
  ],
  "cart:new-total": [
    {
      "type": 0,
      "value": "New Cart Total:"
    }
  ],
  "cart:no-items": [
    {
      "type": 0,
      "value": "There are no items in your shopping cart"
    }
  ],
  "cart:quantity": [
    {
      "type": 0,
      "value": "Quantity"
    }
  ],
  "cart:remove-coupon": [
    {
      "type": 0,
      "value": "Remove coupon"
    }
  ],
  "cart:remove-item": [
    {
      "type": 0,
      "value": "Remove item"
    }
  ],
  "cart:shopping-cart": [
    {
      "type": 0,
      "value": "Shopping Cart"
    }
  ],
  "cart:total": [
    {
      "type": 0,
      "value": "Cart Total:"
    }
  ],
  "cart:update-cart": [
    {
      "type": 0,
      "value": "Update Cart"
    }
  ],
  "checkout:add-billing-information": [
    {
      "type": 0,
      "value": "Add Billing Information"
    }
  ],
  "checkout:add-shipping-information": [
    {
      "type": 0,
      "value": "Add Shipping Information"
    }
  ],
  "checkout:address-city": [
    {
      "type": 0,
      "value": "City"
    }
  ],
  "checkout:address-default-address": [
    {
      "type": 0,
      "value": "Make my default address"
    }
  ],
  "checkout:address-email": [
    {
      "type": 0,
      "value": "E-Mail"
    }
  ],
  "checkout:address-firstname": [
    {
      "type": 0,
      "value": "First Name"
    }
  ],
  "checkout:address-form-address-select-new-address": [
    {
      "type": 0,
      "value": "New Address"
    }
  ],
  "checkout:address-form-heading": [
    {
      "type": 0,
      "value": "Shipping Address"
    }
  ],
  "checkout:address-lastname": [
    {
      "type": 0,
      "value": "Last Name"
    }
  ],
  "checkout:address-phone": [
    {
      "type": 0,
      "value": "Phone"
    }
  ],
  "checkout:address-postcode": [
    {
      "type": 0,
      "value": "ZIP"
    }
  ],
  "checkout:address-save-in-address-book": [
    {
      "type": 0,
      "value": "Save in address book"
    }
  ],
  "checkout:address-state": [
    {
      "type": 0,
      "value": "State"
    }
  ],
  "checkout:address-street": [
    {
      "type": 0,
      "value": "Street"
    }
  ],
  "checkout:address-submit": [
    {
      "type": 0,
      "value": "Use Address"
    }
  ],
  "checkout:address-use-saved-address": [
    {
      "type": 0,
      "value": "Use Saved Address"
    }
  ],
  "checkout:back-to-cart": [
    {
      "type": 0,
      "value": "Back to cart"
    }
  ],
  "checkout:checkout": [
    {
      "type": 0,
      "value": "Checkout"
    }
  ],
  "checkout:confirm-order": [
    {
      "type": 0,
      "value": "Confirm Order"
    }
  ],
  "checkout:continue-shopping": [
    {
      "type": 0,
      "value": "Continue Shopping"
    }
  ],
  "checkout:country": [
    {
      "type": 0,
      "value": "Country"
    }
  ],
  "checkout:order-confirmation": [
    {
      "type": 0,
      "value": "The order number is "
    },
    {
      "type": 1,
      "value": "orderId"
    },
    {
      "type": 0,
      "value": ". You will receive an order confirmation email with order status and other details."
    }
  ],
  "checkout:pay-with": [
    {
      "type": 0,
      "value": "Pay With"
    }
  ],
  "checkout:same-as-shipping": [
    {
      "type": 0,
      "value": "Billing address same as shipping address"
    }
  ],
  "checkout:ship-to": [
    {
      "type": 0,
      "value": "Ship To"
    }
  ],
  "checkout:shipping-information": [
    {
      "type": 0,
      "value": "Shipping Information"
    }
  ],
  "checkout:shipping-method": [
    {
      "type": 0,
      "value": "Shipping Method"
    }
  ],
  "checkout:specify-shipping-method": [
    {
      "type": 0,
      "value": "Specify Shipping Method"
    }
  ],
  "checkout:thankyou-for-purchase": [
    {
      "type": 0,
      "value": "Thank you for your purchase!"
    }
  ],
  "checkout:total": [
    {
      "type": 0,
      "value": "TOTAL"
    }
  ],
  "checkout:use": [
    {
      "type": 0,
      "value": "Use"
    }
  ],
  "checkout:use-card": [
    {
      "type": 0,
      "value": "Use Card"
    }
  ],
  "checkout:use-payment-method": [
    {
      "type": 0,
      "value": "Use Payment Method"
    }
  ],
  "checkout:use-shipping-method": [
    {
      "type": 0,
      "value": "Use Method"
    }
  ],
  "common": [
    {
      "type": 0,
      "value": "Edit"
    }
  ],
  "common:back": [
    {
      "type": 0,
      "value": "Back"
    }
  ],
  "common:cancel": [
    {
      "type": 0,
      "value": "Cancel"
    }
  ],
  "common:close": [
    {
      "type": 0,
      "value": "Close"
    }
  ],
  "common:loading": [
    {
      "type": 0,
      "value": "Loading"
    }
  ],
  "common:save": [
    {
      "type": 0,
      "value": "Save"
    }
  ],
  "product:add-item": [
    {
      "type": 0,
      "value": "Add to Cart"
    }
  ],
  "product:add-to-wishlist": [
    {
      "type": 0,
      "value": "Add to Wish List"
    }
  ],
  "product:customization-price": [
    {
      "type": 0,
      "value": "Your customization"
    }
  ],
  "product:customize-bundle": [
    {
      "type": 0,
      "value": "Customize"
    }
  ],
  "product:gift-card-amount-selection": [
    {
      "type": 0,
      "value": "Amount"
    }
  ],
  "product:gift-card-choose-amount": [
    {
      "type": 0,
      "value": "Choose an Amount"
    }
  ],
  "product:gift-card-custom-ammount-maximum": [
    {
      "type": 0,
      "value": "Maximum"
    }
  ],
  "product:gift-card-custom-ammount-minimum": [
    {
      "type": 0,
      "value": "Minimum"
    }
  ],
  "product:gift-card-custom-amount": [
    {
      "type": 0,
      "value": "Amount in"
    }
  ],
  "product:gift-card-other-amount": [
    {
      "type": 0,
      "value": "Other Amount"
    }
  ],
  "product:quantity-label": [
    {
      "type": 0,
      "value": "Product quantity"
    }
  ],
  "product:required-fields": [
    {
      "type": 0,
      "value": "Required fields"
    }
  ]
}