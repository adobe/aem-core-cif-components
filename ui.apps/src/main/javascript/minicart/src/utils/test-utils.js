/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/

import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';

import QUERY_CART_DETAILS from '../queries/query_cart_details.graphql';

const emptyCartId = 'empty';
const mockCartId = '123ABC';

const mocks = [
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: emptyCartId
            }
        },
        result: {
            data: {
                cart: {
                    email: null,
                    shipping_addresses: [],
                    prices: {
                        grand_total: {
                            currency: 'USD',
                            value: 0
                        }
                    },
                    selected_payment_method: {
                        code: '',
                        title: ''
                    },
                    billing_address: {
                        city: null,
                        country: {
                            code: null
                        },
                        lastname: null,
                        firstname: null,
                        region: {
                            code: null
                        },
                        street: [''],
                        postcode: null,
                        telephone: null
                    },
                    available_payment_methods: [
                        {
                            code: 'cashondelivery',
                            title: 'Cash On Delivery'
                        },
                        {
                            code: 'banktransfer',
                            title: 'Bank Transfer Payment'
                        },
                        {
                            code: 'checkmo',
                            title: 'Check / Money order'
                        },
                        {
                            code: 'free',
                            title: 'No Payment Information Required'
                        }
                    ],
                    items: []
                }
            }
        }
    },
    {
        request: {
            query: QUERY_CART_DETAILS,
            variables: {
                cartId: mockCartId
            }
        },
        result: {
            data: {
                prices: {
                    grand_total: {
                        currency: 'USD',
                        value: 0
                    }
                },
                items: []
            }
        }
    }
];

const AllProviders = ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            {children}
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options) => render(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
