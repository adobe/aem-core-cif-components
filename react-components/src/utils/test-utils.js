/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { I18nextProvider } from 'react-i18next';

import i18n from '../../__mocks__/i18nForTests';
import ConfigContextProvider from '../context/ConfigContext';
import UserContextProvider from '../context/UserContext';

import mutationChangePassword from './mocks/mutationChangePassword';
import mutationCreateCustomer from './mocks/mutationCreateCustomer';
import mutationCreateDuplicateCustomer from './mocks/mutationCreateCustomerDuplicate';
import mutationDeleteCustomerAddress from './mocks/mutationDeleteCustomerAddress';
import mutationGenerateToken from './mocks/mutationGenerateToken';
import mutationGenerateTokenWrongPassword from './mocks/mutationGenerateTokenWrongPassword';
import mutationMergeCarts from './mocks/mutationMergeCarts';
import mutationPlaceOrder from './mocks/mutationPlaceOrder';
import mutationRevokeToken from './mocks/mutationRevokeToken';
import mutationShippingAddress from './mocks/mutationShippingAddress';
import queryCart from './mocks/queryCart';
import queryCountries from './mocks/queryCountries';
import queryCustomerCart from './mocks/queryCustomerCart';
import queryCustomerDetails from './mocks/queryCustomerDetails';
import queryCustomerInformation from './mocks/queryCustomerInformation';
import queryEmptyCart from './mocks/queryEmptyCart';
import queryNewCart from './mocks/queryNewCart';

const defaultMocks = [
    mutationChangePassword,
    mutationCreateCustomer,
    mutationCreateDuplicateCustomer,
    mutationDeleteCustomerAddress,
    mutationGenerateToken,
    mutationGenerateTokenWrongPassword,
    mutationMergeCarts,
    mutationPlaceOrder,
    mutationRevokeToken,
    mutationShippingAddress,
    queryCart,
    queryCountries,
    queryCustomerCart,
    queryCustomerDetails,
    queryCustomerInformation,
    queryEmptyCart,
    queryNewCart
];

const defaultConfig = {
    storeView: 'default',
    graphqlEndpoint: 'none'
};

// eslint-disable-next-line react/display-name
const allProviders = (config, userContext, mocks) => ({ children }) => {
    return (
        <MockedProvider mocks={mocks} addTypename={false}>
            <ConfigContextProvider config={config || defaultConfig}>
                <UserContextProvider initialState={userContext}>
                    <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
                </UserContextProvider>
            </ConfigContextProvider>
        </MockedProvider>
    );
};

/* Wrap all the React components tested with the library in a mocked Apollo provider */
const customRender = (ui, options = {}) => {
    const { config, userContext, mocks = defaultMocks, ...renderOptions } = options;
    return render(ui, { wrapper: allProviders(config, userContext, mocks), ...renderOptions });
};

export * from '@testing-library/react';
export { customRender as render };
