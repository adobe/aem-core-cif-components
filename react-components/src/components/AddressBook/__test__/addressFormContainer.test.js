/*******************************************************************************
 *
 *    Copyright 2019 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { render } from 'test-utils';

import UserContextProvider from '../../../context/UserContext';
import i18n from '../../../../__mocks__/i18nForTests';

import AddressFormContainer from '../addressFormContainer';

describe('<AddressFormContainer>', () => {
    it('renders the component', () => {
        const { asFragment } = render(
            <I18nextProvider i18n={i18n}>
                <UserContextProvider>
                    <AddressFormContainer />
                </UserContextProvider>
            </I18nextProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders the component with address form shown', () => {
        const { asFragment } = render(
            <I18nextProvider i18n={i18n}>
                <UserContextProvider initialState={{ isShowAddressForm: true }}>
                    <AddressFormContainer />
                </UserContextProvider>
            </I18nextProvider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
