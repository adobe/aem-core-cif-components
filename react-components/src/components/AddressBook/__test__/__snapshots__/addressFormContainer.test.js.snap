// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressFormContainer> renders the component 1`] = `
<DocumentFragment>
  <div
    class="mask"
  />
</DocumentFragment>
`;

exports[`<AddressFormContainer> renders the component with address form shown 1`] = `
<DocumentFragment>
  <div
    class="mask_active"
  />
  <div
    class="container"
  >
    <form
      class="root"
    >
      <div
        class="body"
      >
        <h2
          class="heading"
        >
          Address
        </h2>
        <div
          class="firstname"
        >
          <div
            class="root"
          >
            <label
              for="firstname"
            >
              <span
                class="label"
              >
                First Name
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <input
                  class="input"
                  id="firstname"
                  name="firstname"
                  value=""
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              />
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="lastname"
        >
          <div
            class="root"
          >
            <label
              for="lastname"
            >
              <span
                class="label"
              >
                Last Name
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <input
                  class="input"
                  id="lastname"
                  name="lastname"
                  value=""
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              />
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="street0"
        >
          <div
            class="root"
          >
            <label
              for="street0"
            >
              <span
                class="label"
              >
                Street
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <input
                  class="input"
                  id="street0"
                  name="street0"
                  value=""
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              />
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="city"
        >
          <div
            class="root"
          >
            <label
              for="city"
            >
              <span
                class="label"
              >
                City
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <input
                  class="input"
                  id="city"
                  name="city"
                  value=""
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              />
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="country"
        >
          <div
            class="root"
          >
            <label
              for="country"
            >
              <span
                class="label"
              >
                Country
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <select
                  class="input"
                  id="country"
                  name="country_code"
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              >
                <span
                  class="root"
                >
                  <svg
                    fill="none"
                    height="18"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="18"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="6 9 12 15 18 9"
                    />
                  </svg>
                </span>
              </span>
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="region_code"
        >
          <div
            class="root"
          >
            <label
              for="region_code"
            >
              <span
                class="label"
              >
                State
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <input
                  class="input"
                  id="region_code"
                  name="region_code"
                  value=""
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              />
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="postcode"
        >
          <div
            class="root"
          >
            <label
              for="postcode"
            >
              <span
                class="label"
              >
                ZIP
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <input
                  class="input"
                  id="postcode"
                  name="postcode"
                  value=""
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              />
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="telephone"
        >
          <div
            class="root"
          >
            <label
              for="telephone"
            >
              <span
                class="label"
              >
                Phone
              </span>
            </label>
            <span
              class="root"
            >
              <span
                class="input"
              >
                <input
                  class="input"
                  id="telephone"
                  name="telephone"
                  value=""
                />
              </span>
              <span
                class="before"
              />
              <span
                class="after"
              />
            </span>
            <p
              class="root"
            />
          </div>
        </div>
        <div
          class="default_shipping"
        >
          <label
            class="root"
            for="default_shipping"
          >
            <span
              class="icon"
            />
            <input
              class="input"
              id="default_shipping"
              name="default_shipping"
              type="checkbox"
            />
            <span
              class="label"
            >
              Make my default address
            </span>
          </label>
          <p
            class="root"
          />
        </div>
        <div
          class="validation"
        />
        <div
          class="error"
        />
      </div>
      <div
        class="footer"
      >
        <button
          class="root_normalPriority"
          type="button"
        >
          <span
            class="content"
          >
            Cancel
          </span>
        </button>
        <button
          class="root_highPriority"
          type="submit"
        >
          <span
            class="content"
          >
            Save
          </span>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
